### ISSUE TITLE
Under stress, some connections keep open

### ISSUE THREAD

**Issue by revington** [findhit/proxywrap#15]
Under stress some connections will stay on CLOSE_WAIT state.

To reproduce the issue:

`$ node close-wait.js` close-wait.js below
`$ wrk --latency -t4 -c400 -d30s "http://localhost:8000"`

```
$ ss -e | grep 8000 | grep CLOSE
tcp    CLOSE-WAIT 0      0            127.0.0.1:8000          127.0.0.1:40084    uid:1000 ino:2562644 sk:ffff8803a9e2db00 -->
```

Everytime I run the stress test a new connection (the first one) remains in CLOSE_WAIT state

close-wait.js
```
'use strict';
var cluster = require('cluster');
var http = require('http');
var numCPUs = require('os').cpus().length;
var proxiedHttp = require('findhit-proxywrap').proxy(http, {
    strict: false
});
var i;

if (cluster.isMaster) {
    // Fork workers.
    for (i = 0; i < numCPUs; i++) {
        cluster.fork();
    }

} else {

    // Workers can share any TCP connection
    // In this case it is an HTTP server
    proxiedHttp.createServer(function handler(req, res) {
        return res.end('ok');
    }).listen(8000);
}
```

**Comment by cusspvz** [comment]
Interesting... Is this test code intermittent or actually works every time?

We aren't using this at the time on production, but yet I could check it by today. ;)

**Comment by revington** [comment]
It works everytime for me

**Comment by revington** [comment]
wrk, makes a connection before performing the test (the test use another connection/s).
The behavior of node to this connection is different when http has been proxied with proxywrap.

Without proxywrap, node will (record no. 10) answer FIN ACK to wrk
![without-proxywrap](https://cloud.githubusercontent.com/assets/91995/12263940/4c06c866-b933-11e5-9e6d-b3ba00dc5ac9.png)

With proxywrap, node will just (record no. 10) answer only ACK to wrk
![with-proxywrap](https://cloud.githubusercontent.com/assets/91995/12264150/e93f3cde-b934-11e5-85b4-17066161b61d.png)

**Comment by cusspvz** [comment]
I've released `0.3.11`, which includes your fix! Thanks once again.

**Comment by revington** [comment]
You're welcome!

2016-01-18 15:54 GMT+01:00 José Moreira <notifications@github.com>:
